Cautionary alerts residents

benbaumann95 opened the pull request:
### Description of change

- Add Tenancy Information client
- Use property reference as parameter in Tenancy Information Platform API to retrieve the tenancy agreement reference (tag_ref) so that we can use this as parameter on the cautionary alerts residents endpoint
- Build alerts on property json response
- Use HACT naming for alert keys
- Add resident and address alerts to cautionary-alerts endpoint

### Story Link

https://trello.com/c/zD2Gt1sd/124-as-a-user-i-need-to-see-cautionary-alerts-on-residents-in-the-property-so-that-i-can-keep-our-operatives-safe

benbaumann95 committed:
Add staging url and token for tenancy information API

benbaumann95 committed:
Get cautionary alerts on residents
- Add Tenancy Information client
- Use property reference as parameter in Tenancy Information Platform API to retrieve the tenancy agreement reference (tag_ref) so that we can use this as parameter on the cautionary alerts residents endpoint
- Build alerts on property json response
- Use HACT naming for alert keys
- Add resident and address alerts to cautionary-alerts endpoint

neilvanbeinum commented on app/controllers/api/v2/cautionary_alerts_controller.rb:
```diff
@@ -4,7 +4,7 @@ module Api
     class CautionaryAlertsController < ApplicationController
       def index
-        @property = CautionaryAlert.for_property_reference(params[:property_id])
+        @property = BuildCautionaryAlertsService.new(params[:property_id]).perform
```
Might be worth a quick rename of `@property` to avoid confusion  ... `@cautionary_alerts` ?

EGiataganas replied:
How about moving the `BuildCautionaryAlertsService.new(params[:property_id]).perform` into `CautionaryAlerts` class?

benbaumann95 replied:
Hey how exactly did you mean by this @EGiataganas https://github.com/LBHackney-IT/repairs-api/pull/20#discussion_r540955599 ?

neilvanbeinum commented on app/services/build_cautionary_alerts_service.rb:
```diff
@@ -0,0 +1,35 @@
+  end
+
+  # The property exists, but the alerts API has returned a 404, so we return
+  # an array for consistency
```
Please can we move to line 29

EGiataganas commented on app/services/property_search.rb:
```diff
@@ -16,16 +16,12 @@ def perform
     {
       property: property,
-      cautionaryAlerts: cautionary_alerts
+      alert: cautionary_alerts
```
I think this might worth named as `alerts` since it will be a hash of `locationAlerts` and `personAlerts`.

EGiataganas commented on app/models/cautionary_alert.rb:
```diff
@@ -3,20 +3,12 @@
   include ActiveModel::Model
 
-  attr_accessor :propertyReference, :alerts
+  attr_accessor :locationAlert, :personAlert
```
how about use plural here?

benbaumann95 replied:
Yeah I think that would make sense but the only reason I didn't do this is because of HACT naming. I'm not sure whether it is supposed to be pluralised based on the data model sheet. But happy to pluralise if this is compliant with HACT - i'll need to double check

EGiataganas commented on spec/services/platform_apis/cautionary_alerts/client_spec.rb:
```diff
@@ -2,18 +2,26 @@
+      described_class.get_alerts_by_tenancy_reference("01111/01")
     end
   end
 end
```
you forgot to rename the file `spec/services/platform_apis/cautionary_alerts/client_spec.rb` to `spec/services/platform_apis/alerts/client_spec.rb`

benbaumann95 committed:
Return empty array when no tenancy agreement ref is found
- The tenancy information platform API always returns a 200 response but response["tenancies"] can be empty so no tenancy agreement reference is found. This means that there won't be any person alerts on the property. To be consistent we just return an empty array in this scenario
